src/ibex_bluesky_core/devices/simpledae/reducers.py

```diff
@@ -41,15 +44,94 @@ async def sum_spectra(spectra: Collection[DaeSpectra]) -> sc.Variable | sc.DataA
     return summed_counts
 
 
+def tof_bounded_spectra(
+    bounds: sc.Variable,
+) -> Callable[[Collection[DaeSpectra]], Awaitable[sc.Variable | sc.DataArray]]:
+    """Sum a set of neutron spectra between the specified time of flight bounds.
+
+    Args:
+        bounds: A scipp array of size 2, no variances, unit of us,
+            where the second element must be larger than the first.
+
+    Returns a scipp scalar, which has .value and .variance properties for accessing the sum
+    and variance respectively of the summed counts.
+
+    More info on scipp arrays and scalars can be found here: https://scipp.github.io/generated/functions/scipp.scalar.html
+
+    """
+    if "tof" not in bounds.dims:
+        raise ValueError("Should contain tof dims")
+    if bounds.sizes["tof"] != 2:
+        raise ValueError("Should contain lower and upper bound")
+
+    async def sum_spectra_with_tof(spectra: Collection[DaeSpectra]) -> sc.Variable | sc.DataArray:
+        """Sum spectra bounded by a time of flight upper and lower bound."""
+        summed_counts = sc.scalar(value=0, unit=sc.units.counts, dtype="float64")
+        for spec in asyncio.as_completed([s.read_spectrum_dataarray() for s in spectra]):
+            tof_bound_spectra = await spec
+            summed_counts += tof_bound_spectra.rebin({"tof": bounds}).sum()
+        return summed_counts
+
+    return sum_spectra_with_tof
+
+
+def wavelength_bounded_spectra(
+    bounds: sc.Variable, beam_total: sc.Variable
+) -> Callable[[Collection[DaeSpectra]], Awaitable[sc.Variable | sc.DataArray]]:
+    """Sum a set of neutron spectra between the specified [tof/wavelength] bounds.
+
+    Args:
+        bounds: A scipp array of size 2 of wavelength bounds, in units of angstrom,
+            where the second element must be larger than the first.
+        beam_total: A scipp scalar of Ltotal (total flight path length), the path length
+            from neutron source to detector or monitor, in units of meters.
+
+    Time of flight is converted to wavelength using scipp neutron's library function
+        `wavelength_from_tof`, more info on which can be found here:
+        https://scipp.github.io/scippneutron/generated/modules/scippneutron.conversion.tof.wavelength_from_tof.html
+
+    Returns a scipp scalar, which has .value and .variance properties for accessing the sum
+    and variance respectively of the summed counts.
+
+    """
+    if "tof" not in bounds.dims:
+        raise ValueError("Should contain tof dims")
+
+    async def sum_spectra_with_wavelength(
+        spectra: Collection[DaeSpectra],
+    ) -> sc.Variable | sc.DataArray:
+        """Sum spectra and convert time of flight to wavelength."""
```

Comment:
nit: can we make this docstring a little more precise?

e.g.

```suggestion
        """Sum a set of spectra between the specified wavelength bounds."""
```